feat: adds docs skill (#55)
slackapi/slack-mcp-plugin: skills/slack-docs/SKILL.md

---
name: slack-docs

> description: "Search and read the official Slack platform documentation at docs.slack.dev. Use this skill whenever the developer asks how something works in Slack (the Events API, OAuth, Socket Mode, app manifests, incoming webhooks, rate limits, scopes, surfaces like App Home or modals), wants to look up or fetch a specific docs page, pastes a docs.slack.dev link, asks a conceptual or how-to question that is not tied to one specific Web API method, or asks about one of the official tools and SDKs (the Slack CLI, Bolt for JavaScript/Python/Java, the Node/Python/Java Slack SDKs, or the Slack Send GitHub Action). This skill discovers the right page through the docs search API and reads it as raw markdown."

Reviewer: ⚠️ **Description is too broad, causes two tool-selection regressions.**

Two of its triggers already belong to `slack-api`:

- **"pastes a docs.slack.dev link"**: `slack-api`'s description explicitly claims `docs.slack.dev/reference/methods` links. With this broader claim, `skill-slack-api-docs-url` ("What arguments does .../reference/methods/conversations.history take?") now mis-routes to `slack-docs`.
- **"rate limits, scopes"**: `slack-api` owns scope / rate-limit / `missing_scope` questions.

Suggest narrowing to conceptual/guide/how-to + SDK docs, excluding method-reference URLs, and pointing those at `slack:slack-api` explicitly. Proposed rewrite:

```suggestion
description: "Search and read the official Slack platform documentation at docs.slack.dev. Use this skill to answer conceptual or how-to questions about Slack features like the Events API, OAuth, Socket Mode, app manifests, webhooks, modals, and App Home. You can also use it to look up, fetch, or summarize specific guide pages from provided docs.slack.dev links."
```

Author: Ohhh gotcha! didn't know scopes / rate limits fell under that but it make sense

> **Critical rules:**
>
> - The docs are the **source of truth**. Do not answer a factual question about the Slack platform from memory; discover the page, fetch it, then answer from what it says.
> - Prefer the **`.md` version** of any page over the HTML version. It is cleaner for reading and quoting.
> - Every fetched markdown page begins with a `Source: <url>` line. Keep that URL so you can cite the page back to the developer.

> **DO NOT rules:**
>
> - DO NOT invent or guess documentation URLs. Get them from the search API, the sitemap, or a link the developer gave you then verify by fetching.
> - DO NOT paraphrase a page you have not actually fetched. If a fetch fails, say so rather than filling the gap from memory.
> - DO NOT assume every URL has a `.md` version. If a fetch returns an error, fall back to the search API or the sitemap (Step 1) rather than guessing another URL.

---

> ## Fast Path (the developer already has a URL)

Reviewer: 💡 *(Lower priority)* The "drop `#anchor`, append `.md`, WebFetch" logic is duplicated here and in Step 2. Consider making Step 2 the single home for path→`.md`→fetch and having the Fast Path just hand off to it

Goal is to have less to maintain

Author: brilliant. made it so!

If the developer pasted a `https://docs.slack.dev/...` link, skip discovery:

1. Append `.md` to the page path (drop any `#anchor` first), e.g.
   `https://docs.slack.dev/apis/events-api` → `https://docs.slack.dev/apis/events-api.md`.
2. **WebFetch** that `.md` URL and answer from its contents.

The server lowercases `.md` requests, so casing does not matter. `chat.postMessage.md` and `chat.postmessage.md` both resolve.

---

## Step 1: Discover the Page (search)

Use the docs **search API** to find candidate pages. WebFetch (or `curl` via the Bash tool) this URL, with the query URL-encoded:

```
https://docs.slack.dev/api/v1/search?query=<url-encoded query>&limit=5
```

For example, `socket mode` → `https://docs.slack.dev/api/v1/search?query=socket%20mode&limit=5`.

The response is JSON:

```json

> {

Reviewer: 💡 *(Lower priority)*

Verified live: a `socket mode` search actually returns SDK pages (`/tools/java-slack-sdk/…`, `/tools/node-slack-sdk/…`) **above** the events-api guide shown here. 🤔 

Claude claims "The example oversells the ranking.", we might need to improve the search endpoint 😅

Author: maybe quick fix is to have it filter to guides - there is a filter param in endpoint

Author: jk i made a filter on site but not on endpoint hmmmm

Author: no longer jk, aded filter to endpoint and then told it to filter to guides

  "total_results": 98,
  "results": [
    { "url": "/apis/events-api/using-socket-mode", "title": "Using Socket Mode" }
  ],
  "limit": 5
}
```

Scan `results` for the best `title`/`url` match, then read it in **Step 2**. A query is **required**. Calling the endpoint with no `query` returns a `400` with an `error` field.

**Fallbacks** when search does not surface a good hit, or returns a `500`/temporary error (the endpoint is rate-limited and cached ~5 minutes):

- WebFetch `https://docs.slack.dev/llms-sitemap.md`, a flat list of every documentation page's `.md` URL, and scan it for the relevant path.
- For API reference lookups, the enriched index pages are often faster: `https://docs.slack.dev/reference/methods.md`, `.../events.md`, `.../scopes.md`, `.../objects.md`, and `.../block-kit.md` each list every item with a one-line description and a `.md` link.

> - If the developer has the Slack CLI, `slack docs search "<query>"` does the same discovery from the terminal (see the `slack-developer:slack-cli` skill).

Reviewer: Broken cross-reference (You found this in the other changes 🙏 🚀 )

```suggestion
- If the developer has the Slack CLI, `slack docs search "<query>"` does the same discovery from the terminal (see the `slack:slack-cli` skill).
```

---

## Step 2: Read the Page (fetch markdown)

> 

Reviewer: 🤔 `slack-cli` vs `slack-docs` overlap 

Needs a boundary decision (the `skill-slack-cli-socket-mode` failure). We previously expected `slack-cli` for "Search the Slack developer documentation...", but `slack-docs` is arguably the better pick

`slack-cli`'s own description still claims "searching the Slack developer documentation for any topic (socket mode, the Events API, OAuth, …)", which is now redundant with this skill.

Maybe we should move general doc-search to `slack-docs` and narrow the `slack-cli`'s description to "search docs from the terminal via `slack docs search`". Then update the `skill-slack-cli-socket-mode` scenario's `expected_tool` to `slack-docs`.

Author: Narrowed it and updated scenario!

---
name: slack-docs

> description: "Search and read the official Slack platform documentation at docs.slack.dev. Use this skill to answer conceptual or how-to questions about Slack features like the Events API, OAuth, Socket Mode, app manifests, webhooks, modals, and App Home. You can also use it to look up, fetch, or summarize specific guide pages from provided docs.slack.dev links."

Reviewer: I think simpler is better here since what we include on the docs page changes over time

```suggestion
description: "Search and read the official Slack platform documentation at docs.slack.dev. Use this skill to answer conceptual or how-to questions about Slack features. You can also use it to look up, fetch, or summarize specific guide pages from provided docs.slack.dev links."
```

---

# Slack Platform Documentation

Help the developer **find** the right page on the official Slack documentation site (`https://docs.slack.dev`) and **read** it as clean markdown, so answers come from the live docs rather than memory. The site exposes three machine-readable surfaces an agent can use directly, with no authentication and no Slack workspace:

- A **search API method**: `GET https://docs.slack.dev/api/v1/search?query=<q>&category=<c>` returns ranked page hits as JSON. Scope every search with a `category` (guides, reference, a specific SDK, etc.); an uncategorized search skews heavily toward SDK pages.
- **Per-page markdown**: every page is available at its URL **+ `.md`** (e.g. `/quickstart.md`).
- **Index files**: `/llms.txt` (a curated overview) and `/llms-sitemap.md` (a list of every markdown page).

If `$0` is provided, it is either a `docs.slack.dev` URL (jump to the **Fast Path**) or a topic to search (start at **Step 1**). For a broad "how do I build X on Slack?" question rather than one specific page, read `https://docs.slack.dev/llms.txt` first; it is a curated, LLM-oriented overview of the platform and the recommended build path.

> **Critical rules:**
>

> > - The docs are the **source of truth**. Do not answer a factual question about the Slack platform from memory; discover the page, fetch it, then answer from what it says.

Reviewer: Praise 🙏

> - Prefer the **`.md` version** of any page over the HTML version. It is cleaner for reading and quoting.
> - Every fetched markdown page begins with a `Source: <url>` line. Keep that URL so you can cite the page back to the developer.

> **DO NOT rules:**
>

> > - DO NOT invent or guess documentation URLs. Get them from the search API, the sitemap, or a link the developer gave you then verify by fetching.

Reviewer: ```suggestion
> - DO NOT invent documentation URLs. Get them from the search API, the sitemap, or a link the developer gave you then verify by fetching.
```

> - DO NOT paraphrase a page you have not actually fetched. If a fetch fails, say so rather than filling the gap from memory.
> - DO NOT assume every URL has a `.md` version. If a fetch returns an error, fall back to the search API or the sitemap (Step 1) rather than guessing another URL.

---

## Fast Path (the developer already has a URL)

If the developer pasted a `https://docs.slack.dev/...` link, skip discovery and go straight to **Step 2** to fetch and read it.

---

## Step 1: Discover the Page (search)

> Use the docs **search API** to find candidate pages. WebFetch (or `curl` via the Bash tool) this URL, with the query URL-encoded and a `category` to scope the results:

Reviewer: Simpler is better?

```suggestion
Use the docs **search API** to find candidate pages. WebFetch this URL, with the query URL-encoded and a `category` to scope the results:
```

- **`guides`** — the default for "how do I…", "what is…", and conceptual platform questions (Events API, OAuth, Socket Mode, manifests, modals, App Home). Start here when in doubt.
- **`reference`** — for a specific method, event, scope, object, or Block Kit element, especially an exact name like `chat.postMessage`.
- **A tool/SDK category** (`python`, `javascript`, `java`, `slack_cli`, `slack_github_action`, `deno_slack_sdk`) — only once you know the developer's tool. See **Step 3**.

So `socket mode` as a concept → `https://docs.slack.dev/api/v1/search?query=socket%20mode&category=guides&limit=5`.

The response is JSON:

```json
{
  "total_results": 12,
  "results": [
    { "url": "/apis/events-api/using-socket-mode", "title": "Using Socket Mode" }
  ],
  "limit": 5
}
```

Scan `results` for the best `title`/`url` match, then read it in **Step 2**. A query is **required**; calling the endpoint with no `query` returns a `400` with an `error` field.

> Full set of categories:

Reviewer: Is there an endpoint or markdown page that lists these categories rather then hard coding them here?

Author: oh hmmmm there isn't. i can work on adding one but for now i think this is okay because they won't really be changing

Reviewer: Ok no worries, so this is the current source of truth 🙏
